src/anyio/_backends/_asyncio.py

```diff
@@ -600,7 +612,10 @@ def task_done(_task: asyncio.Task) -> None:
             try:
                 exc = _task.exception()
             except CancelledError as e:
-                exc = e
+                if isinstance(e.__context__, CancelledError):
+                    exc = e.__context__
+                else:
+                    exc = e
```

Comment:
```
exc = e
while isinstance(exc.__context__, CancelledError):
    exc = e.__context__
```
I think you want the bottom most CancelledError ?

Reply:
In what situations would we need to dig deeper?

Reply:
the message is in the bottom-most exception: https://bugs.python.org/msg368993